crates/air-lib/trace-handler/src/merger/position_mapping.rs

```diff
@@ -27,19 +26,23 @@ pub(super) enum PreparationScheme {
 pub(super) fn prepare_positions_mapping(scheme: PreparationScheme, data_keeper: &mut DataKeeper) {
     use PreparationScheme::*;
 
-    // it's safe to sub 1 from positions iff scheme was set correctly
-    let prev_pos = match scheme {
-        Previous | Both => Some(data_keeper.prev_slider().position() - 1),
-        Current => None,
-    };
+    let new_pos = data_keeper.result_trace_next_pos();
 
-    let current_pos = match scheme {
-        Current | Both => Some(data_keeper.current_slider().position() - 1),
-        Previous => None,
+    // it's safe to sub 1 from positions here iff scheme was set correctly
+    match scheme {
+        Previous => {
+            let prev_pos = data_keeper.prev_slider().position() - 1;
+            data_keeper.new_to_prev_pos.insert(new_pos, prev_pos);
```

Comment:
There is `BiHashMap::insert_no_overwrite` method.  I suppose values are never overwritten, neither left no right, so shouldn't we use this method with `.expect(...)` to catch bugs earlier?

Reply:
We could do this yes, but in a separate PR, there are some hardening for trace handler that I wanted to do. Moreover, we could do it with this `insert` method since it returns overwrittening status.